specifyweb/specify/migrations/0003_cotype_picklist.py

```diff
@@ -0,0 +1,86 @@
+from django.db import migrations
+
+PICKLIST_NAME = 'CollectionObjectType'
+FIELD_NAME = 'collectionObjectType'
+COTYPE_TEXT = 'Collection Object Type'
+
+def create_cotype_picklist(apps):
+    Collection = apps.get_model('specify', 'Collection')
+    Picklist = apps.get_model('specify', 'Picklist')
+    # Create a cotype picklist for each collection
+    for collection in Collection.objects.all():
+        Picklist.objects.get_or_create(
+            name=PICKLIST_NAME,
+            issystem=True,
+            readonly=True,
+            sizelimit=-1,
+            sorttype=1,
+            type=1,
+            tablename='collectionobjecttype',
+            collection=collection,
+            formatter=PICKLIST_NAME
+        )
+
+def revert_cotype_picklist(apps):
+    Picklist = apps.get_model('specify', 'Picklist')
+    Picklist.objects.filter(name=PICKLIST_NAME).delete()
+
+
+def create_cotype_splocalecontaineritem(apps):
+    Splocalecontainer = apps.get_model('specify', 'Splocalecontainer')
+    Splocalecontaineritem = apps.get_model('specify', 'Splocalecontaineritem')
+    # Create a Splocalecontaineritem record for each CollectionObject Splocalecontainer
+    # NOTE: Each discipline has its own CollectionObject Splocalecontainer
+    for container in Splocalecontainer.objects.filter(name='collectionobject', schematype=0):
+        Splocalecontaineritem.objects.get_or_create(
+            name=FIELD_NAME,
+            picklistname=PICKLIST_NAME,
+            type='ManyToOne',
+            container=container,
+            isrequired=True
+        )
+
+def revert_cotype_splocalecontaineritem(apps):
+    Splocalecontaineritem = apps.get_model('specify', 'Splocalecontaineritem')
+    Splocalecontaineritem.objects.filter(name=FIELD_NAME).delete()
```

Comment:
(could probably be optional)

While _**far**_ less likely then the SpLocaleItemStr case mentioned above,  if we _really_ want to ensure that we are only deleting SpLocaleContainerItem objects fetched/created by these changes, we also have make sure their respective `SpLocaleContainer` do not have the WorkBench schemaType (1) and that the SpLocaleContainerItems are only on the CollectionObject table. 

In case you need it, you can use [field lookups ](https://docs.djangoproject.com/en/5.0/ref/models/querysets/#field-lookups) to make your life a little easier here (and with other comments in this review)!
As an example: 
```py
# queries all SpLocaleContainerItem objects which exist in the (non-workbench) 'collectionobject' container
Splocalecontaineritem.objects.filter(container__name='collectionobject', container__schematype=0)
```

And a more complicated example (using [Q objects](https://docs.djangoproject.com/en/5.1/topics/db/queries/#complex-lookups-with-q-objects)): 
```py
# queries all SpLocaleItemStr objects in the (non-workbench) 'collectionobject' container
Splocaleitemstr.objects.filter(Q(itemdesc__container__name='collectionobject', itemdesc__container__schematype=0) | Q(itemname__container__name='collectionobject', itemname__container__schematype=0))
```